config/params/config.go

```diff
@@ -400,46 +406,47 @@ func (b *BeaconChainConfig) TargetBlobsPerBlock(slot primitives.Slot) int {
 	return b.DeprecatedMaxBlobsPerBlock / 2
 }
 
-// MaxBlobsPerBlock returns the maximum number of blobs per block for the given slot.
 func (b *BeaconChainConfig) MaxBlobsPerBlock(slot primitives.Slot) int {
 	epoch := primitives.Epoch(slot.DivSlot(b.SlotsPerEpoch))
 
-	if epoch >= b.FuluForkEpoch {
-		return b.DeprecatedMaxBlobsPerBlockFulu
+	if len(b.BlobSchedule) > 0 {
+		// Assume BlobSchedule is already sorted ascending by Epoch
+		for i := len(b.BlobSchedule) - 1; i >= 0; i-- {
+			if epoch >= b.BlobSchedule[i].Epoch {
+				return int(b.BlobSchedule[i].MaxBlobsPerBlock)
+			}
+		}
 	}
 
-	if epoch >= b.ElectraForkEpoch {
-		return b.DeprecatedMaxBlobsPerBlockElectra
-	}
-
-	return b.DeprecatedMaxBlobsPerBlock
-}
-
-// MaxBlobsPerBlockByVersion returns the maximum number of blobs per block for the given fork version
-func (b *BeaconChainConfig) MaxBlobsPerBlockByVersion(v int) int {
-	if v >= version.Fulu {
+	switch {
+	case epoch >= b.FuluForkEpoch:
 		return b.DeprecatedMaxBlobsPerBlockFulu
-	}
-
-	if v >= version.Electra {
+	case epoch >= b.ElectraForkEpoch:
```

Comment:
What's the advantage to go from an `if` pattern to a `switch` pattern here?
As a personal taste, I find the `if` pattern clearer.

Reply:
Also, do we still need those Deneb/Electra cases if they are already overwritten in blob schedule?
https://github.com/GabrielAstieres/consensus-specs/blob/7c5b431f40b71bedc2cc5a07a16d8be45d8abfe1/configs/mainnet.yaml#L196-L202

Reply:
I think once there are more than two `if` statements, using a switch becomes cleaner. But I don't feel strongly about this nit, so feel free to overrule me if anyone has a strong preference.

I'm also not sure if we still need this, my only concern was someone running the latest Prysm version but loading an old mainnet YAML without a blob schedule, which could cause issues. This acts as a backup safety check that we can remove later if needed.

Reply:
but thinking more even if loads an old yaml, the latest prysm version should still have blob version schedule as default, I think we are fine. Removing them now